src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -6919,6 +6920,230 @@ def create_stack(self, stack: StackRequest) -> StackResponse:
+
+                            satisfied, msg = requirements.is_satisfied_by(
+                                connector=service_connector,
+                                component=component,
+                            )
+
+                            if not satisfied:
+                                raise RuntimeError(
+                                    "Please pick a connector that is "
+                                    "compatible with the component flavor and "
+                                    "try again.."
+                                )
+
+                            if not resource_id:
+                                if service_connector.resource_id:
+                                    resource_id = service_connector.resource_id
+                                elif service_connector.supports_instances:
+                                    raise RuntimeError(
+                                        f"Multiple {resource_type} resources "
+                                        "are available for the selected "
+                                        "connector. Please use a `resource_id` "
+                                        "to configure a "
+                                        f"{resource_type} resource."
+                                    )
+
+                            try:
+                                connector_resources = self.verify_service_connector(
+                                    service_connector_id=service_connector.id,
+                                    resource_type=requirements.resource_type,
+                                    resource_id=resource_id,
+                                )
+                            except (
+                                KeyError,
+                                ValueError,
+                                IllegalOperationError,
+                                NotImplementedError,
+                                AuthorizationException,
+                            ) as e:
+                                raise RuntimeError(
+                                    f"Access to the resource could not be "
+                                    f"verified: {e}"
+                                )
+
+                            resources = connector_resources.resources[0]
+                            if resources.resource_ids:
+                                if len(resources.resource_ids) > 1:
+                                    raise RuntimeError(
+                                        f"Multiple {resource_type} resources are "
+                                        "available for the selected connector. "
+                                        "Please use the a specific resource-id to "
+                                        f"configure a {resource_type} resource."
+                                    )
+                                else:
+                                    resource_id = resources.resource_ids[0]
```

Comment:
Verification isn't necessary and you should avoid doing any type of verification on the service connector at this point. This should be exclusively about registration. Verification is time-consuming and will yield non-deterministic results.

The verification itself should be done before this endpoint is called.